converter/grpc_interceptor.go

```diff
@@ -55,1360 +45,78 @@ type PayloadCodecGRPCClientInterceptorOptions struct {
-			if err := s.process(
-				encode,
-				o.GetFailure(),
-			); err != nil {
-				return err
-			}
-
-		case *historypb.WorkflowUpdateAcceptedEventAttributes:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetInput(),
-			); err != nil {
-				return err
-			}
-
-		case *historypb.WorkflowUpdateCompletedEventAttributes:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetOutput(),
-			); err != nil {
-				return err
-			}
-
-		case *historypb.WorkflowUpdateRejectedEventAttributes:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetFailure(),
-			); err != nil {
-				return err
-			}
-
-		case *interactionpb.Input:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetArgs(),
-				o.GetHeader(),
-			); err != nil {
-				return err
-			}
-
-		case []*interactionpb.Invocation:
-			for _, x := range o {
-				if err := s.process(encode, x); err != nil {
-					return err
-				}
-			}
-
-		case *interactionpb.Invocation:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetInput(),
-			); err != nil {
-				return err
-			}
-
-		case *interactionpb.Output:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetFailure(),
-				o.GetHeader(),
-				o.GetSuccess(),
-			); err != nil {
-				return err
-			}
-
-		case map[string]*querypb.WorkflowQuery:
-			for _, x := range o {
-				if err := s.process(encode, x); err != nil {
-					return err
-				}
-			}
-
-		case *querypb.WorkflowQuery:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetHeader(),
-				o.GetQueryArgs(),
-			); err != nil {
-				return err
-			}
-
-		case map[string]*querypb.WorkflowQueryResult:
-			for _, x := range o {
-				if err := s.process(encode, x); err != nil {
-					return err
-				}
-			}
-
-		case *querypb.WorkflowQueryResult:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetAnswer(),
-			); err != nil {
-				return err
-			}
-
-		case *schedulepb.Schedule:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetAction(),
-			); err != nil {
-				return err
-			}
-
-		case *schedulepb.ScheduleAction:
-			if o == nil {
-				continue
-			}
-			if err := s.process(
-				encode,
-				o.GetStartWorkflow(),
-			); err != nil {
-				return err
-			}
 
-		case []*schedulepb.ScheduleListEntry:
-			for _, x := range o {
-				if err := s.process(encode, x); err != nil {
+	dc := GetDefaultDataConverter()
+	return proxy.NewFailureVisitorInterceptor(proxy.FailureVisitorInterceptorOptions{
+		Outbound: &proxy.VisitFailuresOptions{
+			Visitor: func(vpc *proxy.VisitFailuresContext, failure *failurepb.Failure) error {
+				var err error
+				failure.EncodedAttributes, err = dc.ToPayload(encodedFailure{
+					Message:    failure.Message,
+					StackTrace: failure.StackTrace,
+				})
+				return err
```

Comment:
You also have to set the existing `Message` to `"Encoded failure"` and unset the `StackTrace`.

It may be best if, even though small, you expose a helper from `internal/failure_converter.go` that that and this call to encode/decode common attributes.